train.py

```diff
@@ -335,41 +430,170 @@ def download_weights():
+    if not videos_without_captions:
+        return caption_files
+
+    print("\n=== 🤖 Auto-captioning Videos ===")
+    print(f"Found {len(videos_without_captions)} videos without captions")
+    model, processor = setup_qwen_model()
+
+    new_caption_files = caption_files.copy()
+    for i, vid_name in enumerate(videos_without_captions, 1):
+        mp4_path = os.path.join(videos_path, vid_name + ".mp4")
+        if os.path.exists(mp4_path):
+            print(f"\n[{i}/{len(videos_without_captions)}] 🎥 {vid_name}.mp4")
+
+            # Use absolute path
+            abs_path = os.path.abspath(mp4_path)
+
+            # Build caption components
+            prefix = f"{autocaption_prefix.strip()} " if autocaption_prefix else ""
+            suffix = f" {autocaption_suffix.strip()}" if autocaption_suffix else ""
+            trigger = f"{trigger_word} " if trigger_word else ""
+
+            # Prepare messages format with customized prompt
+            messages = [
+                {
+                    "role": "user",
+                    "content": [
+                        {
+                            "type": "video",
+                            "video": abs_path,
+                        },
+                        {
+                            "type": "text",
+                            "text": "Describe this video clip in detail, focusing on the key visual elements, actions, and overall scene.",
+                        },
+                    ],
+                }
+            ]
+
+            # Process inputs
+            text = processor.apply_chat_template(
+                messages, tokenize=False, add_generation_prompt=True
+            )
+
+            try:
+                # Import qwen utils here to avoid circular imports
+                from qwen_vl_utils import process_vision_info
+
+                image_inputs, video_inputs = process_vision_info(messages)
+
+                inputs = processor(
+                    text=[text],
+                    images=image_inputs,
+                    videos=video_inputs,
+                    padding=True,
+                    return_tensors="pt",
+                ).to("cuda")
+
+                print("\nGenerating caption...")
+                generated_ids = model.generate(
+                    **inputs,
+                    max_new_tokens=128,
+                    do_sample=True,
+                    temperature=0.7,
+                    top_p=0.9,
+                )
+                generated_ids_trimmed = [
+                    out_ids[len(in_ids) :]
+                    for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
+                ]
+                caption = processor.batch_decode(
+                    generated_ids_trimmed,
+                    skip_special_tokens=True,
+                    clean_up_tokenization_spaces=False,
+                )[0]
+
+                # Combine prefix, trigger, caption, and suffix
```

Comment:
very nit - do we want spaces between these? something like `" ".join([prefix, trigger, caption.strip(), suffix])`